src/test/java/org/apache/commons/io/IOUtilsTest.java

```diff
@@ -1643,5 +1644,19 @@ public void testToString_URL_CharsetName() throws Exception {
     public void testToString_URL_CharsetNameNull() throws Exception {
         testToString_URL(null);
     }
+    
+    @Test
+    public void testBigString() throws IOException {
+       String foo = StringUtils.repeat("\uD83D", (Integer.MAX_VALUE)/2-1);
```

Comment:
No shared CI will provide you these resources.

Reply:
> No shared CI will provide you these resources.

Hi @michael-o 
I wrote a different version of the test that is less resource-intensive and more lenient. See git master. The main code is still broken though.